### PR TITLE
Enhance DWARF support with improved function info import and typed signatures

### PR DESCRIPTION
**Pull request by 0verflowme**
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

File: libr/anal/c2/kv.c [hunk 1]
```diff
@@ -90,7 +90,7 @@ static const char kvc_peek(KVCParser *kvc, int delta) { // rename to peek_at
 }
 
 static void kvc_error(KVCParser *kvc, const char *msg) {
-	R_LOG_ERROR ("Parsing problem at line %d: %s", kvc->line, msg);
+	R_LOG_WARN ("Parsing problem at line %d: %s", kvc->line, msg);
```
Comment (trufae) [inline libr/anal/c2/kv.c]: Uhm inho a parsing error must be an error

**Comment by trufae** [comment]
augmentcode review:

# PR #25437 Review — Enhance DWARF support with improved function info import and typed signatures

**PR**: [radareorg/radare2#25437](https://github.com/radareorg/radare2/pull/25437)
**Author**: `0verflowme`
**Changes**: +384 / -106 across 8 files
**Target**: `master`

## Summary

This PR significantly improves DWARF debug info handling by: separating formal parameters from local variables (fixing a long-standing `is_arg=false` bug), generating typed function signatures, importing function types into `sdb_types` via C parsing with a direct-sdb fallback, and adding a new `r_anal_import_c_decls` public API. The changes include good test coverage and several null-safety improvements.

## Verdict: ⚠️ Needs Changes

Two medium-confidence issues found — one likely misplaced API call and one guideline violation. Two low-priority items for consideration.

### Positive Highlights
- ✅ **Bug fix**: Old code always passed `is_arg=false` to `r_anal_function_set_var`, misclassifying all DWARF formal parameters as local variables — now correctly passes `true` for args
- ✅ **Null safety**: New code checks `!var->name` before use, preventing potential crash when a variable has location data but no name
- ✅ **Clean refactoring**: Variable integration logic factored into `integrate_dwarf_var()`, eliminating duplicated parsing code and `goto` labels
- ✅ **Good test coverage**: Both unit tests and r2r integration tests updated to verify arg/var separation and typed signatures
- ✅ **Smart fallback**: C-parser-first approach with direct sdb fallback handles non-C languages (Rust, Go, Ada) gracefully

## Issues

- [ ] [🟠 `r_bin_get_types()` likely returns PE types, not PDB types](intent://local/task/7b9c8892-49f9-4fa9-b345-bb64b4c1548d)

- [ ] [🟠 `R_RETURN_VAL_IF_FAIL` used for malloc check in static function](intent://local/task/581eda21-51a3-4738-b0ce-6bb867254547)

- [ ] [🟡 DWARF force-load workaround in `cmd_aaa` lacks documentation](intent://local/task/aeb100fc-629e-40b0-9e59-c0e7b601abaa)

- [ ] [🟡 Hardcoded "cdecl" calling convention in fallback](intent://local/task/b7f20c7d-b44d-4991-ae95-91c5527fc504)

**Comment by trufae** [comment]
lgtm
